### PR TITLE
feat: initialize setup of Storybook

### PR DESCRIPTION
**Pull request by TylerAPfledderer**
Initializing Storybook in the new NextJS environment. Sets up the primary configuration, Chakra, and i18next.

### CODE REVIEW

File: .storybook/main.ts
Comment (TylerAPfledderer) [inline .storybook/main.ts:21]: Incrementally add component stories as necessary until all are included since there are some that still have gatsby package imports. 

There is more than just the Button files that could be included here. What do you think @pettinarip ?

Comment (pettinarip) [reply .storybook/main.ts:21]: Makes sense @TylerAPfledderer. A similar approach to the one we are taking in the pages' migration.

**Review by pettinarip — APPROVED** [review]
LGTM @TylerAPfledderer

Do you think we should wait until #96 gets merged?

**Comment by TylerAPfledderer** [comment]
> LGTM @TylerAPfledderer
> 
> Do you think we should wait until #96 gets merged?

That will be helpful!

File: package.json
Comment (pettinarip) [inline package.json:78]: Is there any special reason why we need to use this specific version? Noticed that this is a dep of `storybook`

**Review by pettinarip — COMMENTED** [review]
Fixed conflicts and updated branch

**Review by pettinarip — CHANGES_REQUESTED** [review]
@TylerAPfledderer I'm seeing the following error when I run `yarn storybook`
image

File: package.json
Comment (TylerAPfledderer) [reply package.json:78]: There is an issue with another dependency in storybook tied to the use of jackspeak and working with Yarn v1

https://github.com/storybookjs/storybook/issues/22431#issuecomment-1630086092

And I can't add a comment in `package.json` to notate this 😅

**Comment by TylerAPfledderer** [comment]
> @TylerAPfledderer I'm seeing the following error when I run `yarn storybook` image

My response to your question about the jackspeak dep also applies here. Basically, have to regenerate yarn.lock when you switch to this PR.

**Comment by TylerAPfledderer** [comment]
@pettinarip On a separate note:

I would like to address these in a separate PR, but for the button's `MultiLineText`, `ButtonTwoLines` and `Overrides` stories, there is an issue for each:
- With `Multiline` and `ButtonTwoLines`: some of the theming is not appearing for these with proper spacing and padding.
- With `Overrides`: The i18next is not set up yet for translations.

Would like to know how you suggest addressing these while incrementally adding in component stories

File: package.json
Comment (pettinarip) [reply package.json:78]: Oh good to know, thanks for sharing the link. At least we have this comment now explaining in case anyone is curious 😄

**Comment by pettinarip** [comment]
> @pettinarip On a separate note:
> 
> I would like to address these in a separate PR, but for the button's `MultiLineText`, `ButtonTwoLines` and `Overrides` stories, there is an issue for each:
> 
> * With `Multiline` and `ButtonTwoLines`: some of the theming is not appearing for these with proper spacing and padding.
> * With `Overrides`: The i18next is not set up yet for translations.
> 
> Would like to know how you suggest addressing these while incrementally adding in component stories

Not sure if I follow you on this. I'd try to fix those two things that you are pointing out first and then address the rest of the components' stories. Not sure if that answers your question.

**Comment by TylerAPfledderer** [comment]
@pettinarip I was meaning to not bloat this PR, because it has to do with making sure Storybook itself functions in this fork. 

I can take the time to see if I can rectify the theming for the buttons and add the storybook addon for i18 in this PR 😁

**Comment by TylerAPfledderer** [comment]
@pettinarip it ended up being quick fixes.

Should be good with the theming usage in the button stories and added i18next config for storybook

**Review by pettinarip — APPROVED** [review]
Thanks for setting up SB @TylerAPfledderer. Its working perfectly for me now.

**Pull request merged**
